sentry-android/src/main/java/io/sentry/android/AndroidSentryClientFactory.java

```diff
@@ -35,15 +37,34 @@
      * @param ctx Android Context.
      */
     public AndroidSentryClientFactory(Context ctx) {
-        this.ctx = ctx;
-
         Log.d(TAG, "Construction of Android Sentry.");
+
+        this.ctx = ctx.getApplicationContext();
     }
 
     @Override
     public SentryClient createSentryClient(Dsn dsn) {
+        if (!checkPermission(Manifest.permission.INTERNET)) {
+            Log.e(TAG, Manifest.permission.INTERNET + " is required to connect to the Sentry server,"
+                + " please add it to your AndroidManifest.xml");
+        }
+
+        Log.d(TAG, "Sentry init with ctx='" + ctx.toString() + "' and dsn='" + dsn + "'");
+
+        String protocol = dsn.getProtocol();
+        if (!(protocol.equalsIgnoreCase("http") || protocol.equalsIgnoreCase("https"))) {
+            throw new IllegalArgumentException("Only 'http' or 'https' connections are supported in"
+                + " Sentry Android, but received: " + protocol);
+        }
+
+        if ("false".equalsIgnoreCase(dsn.getOptions().get(DefaultSentryClientFactory.ASYNC_OPTION))) {
```

Comment:
The reason is that the right hand side can be `null`, but I agree, I yanked out a string now.